### PR TITLE
explicitly create interpolants from functions

### LINKED ISSUE #2410: Control automatic function parameter interpolation method
### Description

Allow users to specify which interpolation method they would like to be applied to function parameters. 

### Motivation

When using three-dimensional lookup tables I have found better model build times when moving from "cubic" to "linear" interpolation (in particular convert_to_casadi is slow). At the moment I can only control this by modifying the process symbol method of parameter values. 

The speed up might be due to differences in calculating the Jacobian (or perhaps not calculating it if the Jacobian at all is not defined everywhere for linear interpolation) but I am not sure of the exact reason. 

### Possible Implementation

I can think of a couple of ways of doing this but I'm not sure what is preferable for everyone. Both should be non-breaking. 

**1. Pass optional dictionary to parameter_values class** 
This would just involve adding an optional argument to the parameter_values class and then adding the input as an attribute of the class. The user would just pass an options dictionary in the following way: 
```python
options = {"interpolation methods": {"name of function parameter to target": "linear"} 
```
if not specified then "cubic" interpolation will be used so that this change is nonbreaking. 

**2. Allow users to add a third entry to the tuple within the parameter values dictionary** 
i.e. 
```python
options = {"interpolation method": "linear"}
parameter_values_dictionary = {
   "negative electrode open circuit voltage [V]": ("negative electrode open circuit voltage [V]", ( (x_data, ), y_data), options), 
   "positive electrode open circuit voltage [V]": ("positive electrode open circuit voltage [V]", ( (x_data, ), y_data)), 
}
```
This could either be a dictionary like I suggest here (so that additional control can be easily added later) or it could just be "linear" or "cubic". This third argument should be optional with "cubic" used by default so that we don't break all previous code. 

### Additional context

I am happy to implement once there is agreement on how to do so. 

Could perhaps be done in conjunction with modifications to function parameter in #2403

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

In the standard parameter sets, create the interpolants from functions explicitly (instead of relying on the `ParameterValues` class to do it). This allows finer control of hyperparameters (e.g. interpolator).

Using "cubic" interpolator for OCV gives much faster performance than linear interpolator (e.g. 1s -> 0.5s solve time for Ai2020 on my laptop)

Fixes #2410 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: examples/scripts/compare_lithium_ion.py [hunk 1]
```diff
@@ -7,16 +7,17 @@
 
 # load models
 models = [
-    pybamm.lithium_ion.SPM(),
-    pybamm.lithium_ion.SPMe(),
+    # pybamm.lithium_ion.SPM(),
+    # pybamm.lithium_ion.SPMe(),
```
Comment (Scottmar93) [inline examples/scripts/compare_lithium_ion.py:11]: should these be uncommented?

File: pybamm/input/parameters/ecm/example_set.py [hunk 2]
```diff
@@ -5,13 +5,38 @@
...
-dUdT = pybamm.parameters.process_2D_data_csv("ecm_example_dudt.csv", path=path)
+dUdT_data = pybamm.parameters.process_2D_data_csv("ecm_example_dudt.csv", path=path)
+
+
+def ocv(sto):
+    name, (x, y) = ocv_data
+    return pybamm.Interpolant(x, y, sto, name)
+
+
+def r0(T_cell, current, soc):
```
Comment (Scottmar93) [inline pybamm/input/parameters/ecm/example_set.py:22]: A nice, this is better

**Review by Scottmar93 — APPROVED** [review]
Looks great cheers

**Comment by tinosulzer** [comment]
ready for review (changed a bit since @Scottmar93 reviewed)

File: examples/scripts/compare_lithium_ion.py (hunk 1 above)
Comment (brosaplanella) [reply examples/scripts/compare_lithium_ion.py:11]: These are still commented

File: examples/scripts/compare_lithium_ion.py [hunk 3]
```diff
@@ -7,16 +7,18 @@
...
-    pybamm.lithium_ion.NewmanTobias(),
+    # pybamm.lithium_ion.NewmanTobias(),
 ]
 
 # create and run simulations
 sims = []
 for model in models:
-    sim = pybamm.Simulation(model)
+    sim = pybamm.Simulation(
+        model, parameter_values=pybamm.ParameterValues("NCA_Kim2011")
```
Comment (brosaplanella) [inline examples/scripts/compare_lithium_ion.py:20]: Do we want to go back to the default parameters (i.e. not prescribed)?

**Review by brosaplanella — APPROVED** [review]
Looks great, thanks Valentin! Just a couple of very minor comments.

**Pull request merged**